include/IECoreScene/PrimitiveVariable.h

```diff
@@ -102,11 +102,75 @@ struct IECORESCENE_API PrimitiveVariable
 	/// array of unique UVs in data and map them to FaceVarying using
 	/// the indices.
 	IECore::IntVectorDataPtr indices;
+
+	template<typename T>
+	class IndexedRange;
+
+};
+
+/// Utility class for iterating the `data` field from
+/// a PrimitiveVariable, using the `indices` field
+/// appropriately if it exists.
+///
+/// > Note : This intentionally only provides `const`
+/// > access because it does not make sense to modify
+/// > the `data` elements via the `indices`, since
+/// > each element will be visited an unknown number of
+/// > times.
+template<typename T>
+class PrimitiveVariable::IndexedRange
+{
+
+	public :
+
+		/// Throws if the PrimitiveVariable doesn't contain
+		/// `TypedData<vector<T>>`.
+		IndexedRange( const PrimitiveVariable &variable );
+
+		class Iterator;
+
+		Iterator begin();
+		Iterator end();
+
+		const T &operator[]( size_t i ) const
```

Comment:
We use std::vector<bool> for some of our data storage arrays and this will fail to compile.

Perhaps we should do one of the following?
- remove BoolVectorData 
- return by value from operator[] ?

Reply:
Good catch! I've addressed this in 478f9a5, by using `vector<T>::const_reference` for the return type. That means we still return by reference for the regular vector types (important for StringVectorData, and the concept that this is a view), but return by value for BoolVectorData (in the same way that `vector<bool>` already does).